src/core/dht.cc

```diff
@@ -706,7 +706,9 @@ Sp<Task> DHT::findPeer(const Id& id, int expected, LookupOption option, std::fun
     return task;
 }
 
-Sp<Task> DHT::announcePeer(const Id& peerId, int port, std::function<void(std::list<Sp<NodeInfo>>)> completeHandler) {
+Sp<Task> DHT::announcePeer(const Id& peerId, const Id& proxyId, uint16_t port,
+        const std::string alt, const std::vector<std::uint8_t>& signature,
```

Comment:
const std::string alt --> const std::string& alt   
try to be an object reference as input parameter.